server/src/test/java/org/elasticsearch/node/NodeTests.java

```diff
@@ -339,4 +343,55 @@ public void setCircuitBreaker(CircuitBreaker circuitBreaker) {
             myCircuitBreaker.set(circuitBreaker);
         }
     }
+
+    public static class TestRestCompatibility1 extends Plugin implements RestCompatibilityPlugin {
+        @Override
+        public Version getCompatibleVersion(ParsedMediaType acceptHeader, ParsedMediaType contentTypeHeader, boolean hasContent) {
+            return Version.CURRENT.previousMajor();
+        }
+    }
+
+    public static class TestRestCompatibility2 extends Plugin implements RestCompatibilityPlugin {
+        @Override
+        public Version getCompatibleVersion(ParsedMediaType acceptHeader, ParsedMediaType contentTypeHeader, boolean hasContent) {
+            return null;
+        }
+    }
+
+    public void testLoadingMultipleRestCompatibilityPlugins() throws IOException {
+        Settings.Builder settings = baseSettings();
+
+        // throw an exception when two plugins are registered
+        List<Class<? extends Plugin>> plugins = basePlugins();
+        plugins.add(TestRestCompatibility1.class);
+        plugins.add(TestRestCompatibility2.class);
+
+        expectThrows(IllegalStateException.class, () -> new MockNode(settings.build(), plugins));
```

Comment:
can you check the message? We've had tests like this before and didn't check the message and there was some other cause of the exception rather than the true cause.